Parser/asdl_c.py

```diff
@@ -1148,62 +1148,70 @@ def visitModule(self, mod):
     // known AST class fields
     PyObject *fields = NULL;
     // current instance dictionary
-    PyObject *__dict__ = NULL;
+    PyObject *dict = NULL;
     // constructor positional and keyword arguments
     PyObject *empty_tuple = NULL, *payload = NULL;
-    // for iterating over 'fields' and 'kwargs'
-    PyObject *key = NULL, *value = NULL;
 
-    PyTypeObject *type = Py_TYPE(self);
-    if (PyObject_GetOptionalAttr((PyObject *)type, state->_fields, &fields) < 0) {
+    PyObject *type = (PyObject *)Py_TYPE(self);
+    if (PyObject_GetOptionalAttr(type, state->_fields, &fields) < 0) {
         goto cleanup;
     }
-    if (PyObject_GetOptionalAttr(self, state->__dict__, &__dict__) < 0) {
+    if (PyObject_GetOptionalAttr(self, state->__dict__, &dict) < 0) {
         goto cleanup;
     }
-    if (!(empty_tuple = PyTuple_New(0))) {
+    empty_tuple = PyTuple_New(0);
+    if (empty_tuple == NULL) {
         goto cleanup;
     }
-    if (!(payload = PyDict_New())) {
+    payload = PyDict_New();
+    if (payload == NULL) {
         goto cleanup;
     }
     if (fields) {
         Py_ssize_t numfields = PySequence_Size(fields);
         if (numfields == -1) {
             goto cleanup;
         }
+        int rc;
+        // for iterating over 'fields' and 'kwargs'
+        PyObject *key = NULL, *value = NULL;
         for (Py_ssize_t i = 0; i < numfields; i++) {
-            if (!(key = PySequence_GetItem(fields, i))) {
+            key = PySequence_GetItem(fields, i);
+            if (key == NULL) {
                 goto cleanup;
             }
-            if (PyDict_GetItemRef(__dict__, key, &value) < 0) {
+            rc = PyDict_GetItemRef(dict, key, &value);
+            if (rc < 0) {
+                Py_DECREF(key);
                 goto cleanup;
             }
-            if (!value) {
-                break;
+            if (rc == 0) {
+                Py_DECREF(key);
+                Py_DECREF(value);
```

Comment:
`value` is always NULL here.

Actually, you can even use `value == NULL` instead of `rc == 0` if you prefer.

Reply:
Wait, I thought I removed that one.

Reply:
Ah! I directly edited the C-file... which got overwritten by the generator...